### DISCUSSION TITLE
Merge path of nested object

### DISCUSSION THREAD

**Discussion by sschneider-ihre-pvs** [colinhacks/zod#1632] [Q&A] [answered]
given I have a nested z.object that has a property that is again a z.object. Can I unify all errors from the nested object under a single path?
Pseudo example
```ts
const something = z.object({
  a: z.string(),
  b: z.object({
    id: z.number(),
    label: z.string(),
  }).refine((data) => data,{ path: ['b'] })
})
```
So if there are errors in b, the should be thrown with path b and not b.id or b.label

**Comment by JacobWeisenburger** [comment]
Could you explain more about your use case? I'm not really sure why someone would want to do such a thing, and maybe there is a better way to address this instead of trying to mutate the error path.

**Comment by JacobWeisenburger** [comment, marked as answer]
Is this what you are looking for?

```ts
const something = z.object( {
    a: z.string(),
    b: z.object( {
        id: z.number(),
        label: z.string(),
    } )
} )

const result = something.safeParse( { a: 'string', b: { id: '42' } } )
!result.success && console.log( result.error.flatten() )
// {
//     formErrors: [],
//     fieldErrors: { b: [ "Expected number, received string", "Required" ] }
// }
```
